backend/datasource/caching.go

```diff
@@ -0,0 +1,179 @@
+package datasource
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+var (
+	// ErrorNotCached is returned when a cached value was not found in the internal cache
+	ErrorNotCached = errors.New("no cached response found")
```

Comment:
I don't think this would be an error. As non-cache hits are expected, and not errors. Rather, second return bool param, `found` or something.

Reply:
makes sense! I will update that next week.